src/python/zquantum/core/wavefunction.py

```diff
@@ -208,7 +208,7 @@ def flip_wavefunction(wavefunction: Wavefunction):
     return Wavefunction(flip_amplitudes(wavefunction.amplitudes))
 
 
-def flip_amplitudes(amplitudes: np.ndarray) -> np.ndarray:
+def flip_amplitudes(amplitudes: Union[Sequence[complex], np.ndarray]) -> np.ndarray:
```

Comment:
[suggestion] Why not use `StateVector` here?

Reply:
Because importing `StateVector` from `backend.py` causes circular imports because `backend.py`needs to import from `wavefunction.py`. This will be fixed when we move `StateVector` to `typing.py`.